frontend/public/components/routes.tsx

```diff
@@ -307,14 +312,270 @@ const filters = [{
+              <p>The hostname cannot be changed after the route is created.</p>
+              {/* TODO:  add additional wildcard help text from https://github.com/openshift/origin-web-console/blob/master/app/views/directives/osc-routing.html#L65-L67 */}
+            </div>
+          </div>
+          <div className="form-group co-create-route__path">
+            <label htmlFor="path">Path</label>
+            <input className="form-control"
+              type="text"
+              onChange={this.handleChange}
+              value={this.state.stringData.path}
+              placeholder="/"
+              id="path"
+              name="path" />
+            <div className="help-block">
+              Path that the router watches to route traffic to the service.
+            </div>
+          </div>
+          <div className="form-group co-create-route__service">
+            <label className="co-required" htmlFor="service">Service</label>
+            {(loaded && _.isEmpty(serviceOptions)) && <p className="alert alert-info co-create-route__alert"><span className="pficon pficon-info" aria-hidden="true"></span> There are no services in your project to expose with a route.</p> }
+            {(loaded && !_.isEmpty(serviceOptions)) && <Dropdown items={serviceOptions} title="Select service" dropDownClassName="dropdown--full-width" id="service" onChange={this.changeService} /> }
+            <div className="help-block">
+              Service to route to.
+              {/* TODO:  add additional alterate route help text from https://github.com/openshift/origin-web-console/blob/master/app/views/directives/osc-routing-service.html#L16 */}
+            </div>
+          </div>
+          <div className="form-group co-create-route__target-port">
+            <label htmlFor="target-port">Target Port</label>
+            {_.isEmpty(targetPorts) && <p>Select a service above</p>}
+            {!_.isEmpty(targetPorts) && <Dropdown items={targetPorts} title="Select target port" dropDownClassName="dropdown--full-width" id="target-port" onChange={this.changeTargetPort} /> }
+            <div className="help-block">
+              Target port for traffic.
+            </div>
+          </div>
+          {/* TODO: add alternate services */}
+          {/* TODO: add services
+          <h2 className="h3">Security</h2>
+          <div className="checkbox">
+            <label>
+              <input type="checkbox"
+                onChange={this.toggleSecure}
+                checked={this.state.secure}
+                id="secure"
+                name="secure" />
+              Secure route
+            </label>
+            <div className="help-block">
+              Routes can be secured using several TLS termination types for serving certificates.
+            </div>
+          </div>
+          { this.state.secure && <div className="co-create-route__security">
+            security stuff here
+          </div> } */}
+          {/* TODO: add labels */}
```

Comment:
I would just skip label inputs since we don't have them for other editors.

I would copy the labels from the service, though, since that's what `oc expose` does by default.